es-app/src/CollectionSystemManager.cpp

```diff
@@ -819,15 +789,22 @@ SystemData* CollectionSystemManager::createNewCollectionEntry(std::string name,
 	return newSys;
 }
 
-void CollectionSystemManager::addRandomGames(SystemData* newSys, SystemData* sourceSystem, FileData* rootFolder, 
-	FileFilterIndex* index, std::map<std::string, int> settingsValues, int defaultValue) 
+void CollectionSystemManager::addRandomGames(SystemData* newSys, SystemData* sourceSystem, FileData* rootFolder,
+	FileFilterIndex* index, std::map<std::string, std::map<std::string, int>> mapsForRandomColl, int defaultValue)
 {
-	
+
 	int gamesForSourceSystem = defaultValue;
-	if (settingsValues.find(sourceSystem->getFullName()) != settingsValues.end()) 
+	for (auto& m : mapsForRandomColl)
 	{
-		// we won't add more than the max
-		gamesForSourceSystem = Math::min(RANDOM_SYSTEM_MAX, settingsValues[sourceSystem->getFullName()]);
+		auto key = m.first;
```

Comment:
Do we need to do auto? Can we not use the actual type?

Reply:
Same for the one below.

Reply:
Sure thing.